### PR TITLE
DAOS-6338 DAOS:Change DAOS license text #1

### PR DESCRIPTION
**Pull request by GitHuaKuang**
For these directoriesi and their sub_dirs: 
daos/src/vos 
daos/src/vea 
daos/src/utils 
daos/src/tests 
daos/src/security 

Skip-build: true 
Skip-test: true

### CODE REVIEW

**Comment by GitHuaKuang** [comment]
Created this new PR, closed the old PR-4127 due to something messed up at merging the latest master.

After discussion and research, her is the response from John Pavolotsky in Intel Law and Policy Group:

Thanks all. I have reviewed the various documents and cannot find an obligation to include government rights language in the source header. DAOS is mentioned in the assertions and while Apache 2 is the listed license, we have the right to update it. Accordingly, the header can read:

/**
* (C) Copyright 2018-2020 Intel Corporation.
*
* SPDX-License-Identifier: BSD-2-Clause-Patent
*/

File: src/tests/ftest/aggregation/aggregation_basic.py [hunk 1]
```diff
@@ -2,24 +2,7 @@
 """
   (C) Copyright 2020 Intel Corporation.
```
Comment (kjacque) [inline src/tests/ftest/aggregation/aggregation_basic.py:3]: Looks like some of these didn't get hit by the copyright git hook?

**Review by kjacque — APPROVED** [review]
Security headers look good.

File: src/tests/ftest/aggregation/aggregation_basic.py (hunk 1 above)
Comment (GitHuaKuang) [reply src/tests/ftest/aggregation/aggregation_basic.py:3]: You are right, Kris, actually many, if not all .py files header are not properly updated with the year.
Currently the daos/.git/hooks/pre-commit which links to utils/githooks/pre-commit is not able to
handle python files yet, will have to wait for future update when this ticket is solved:
https://jira.hpdd.intel.com/browse/DAOS-5705

Comment (kjacque) [reply src/tests/ftest/aggregation/aggregation_basic.py:3]: Okay, good to know there's already a ticket.

**Review by jolivier23 — COMMENTED** [review]
Do we not need to add changes to LICENSE file in this patch?

**Comment by GitHuaKuang** [comment]
> Do we not need to add changes to LICENSE file in this patch?

Good point, @skirvan44 currently daos/LICENSE is for Apache 2.0 which is https://www.apache.org/licenses/LICENSE-2.0.txt,  not sure what's the official .txt for BSD-2-Clause-Patent, may be the "Text" section in https://spdx.org/licenses/BSD-2-Clause-Patent.html ?

**Comment by GitHuaKuang** [comment]
Updated DAOS top LICENSE file to match description in https://spdx.org/licenses/BSD-2-Clause-Patent.html

**Review by kjacque — APPROVED** [review]

**Comment by GitHuaKuang** [comment]
Will try to apply @dpquigl change in pre-commit to get .py files date updated - in progress.

**Comment by GitHuaKuang** [comment]
Updated .py files Copyright dates, and also with the utils/githooks/pre-commit change for .py handling.

**Review by kjacque — DISMISSED** [review]

**Comment by kjacque** [comment]
Again, just skimmed the changes. But it looks okay to me.

**Review by johannlombardi — APPROVED** [review]

File: LICENSE [hunk 2]
```diff
@@ -1,176 +1,15 @@
-                                 Apache License
-                           Version 2.0, January 2004
-                        http://www.apache.org/licenses/
+https://spdx.org/licenses/BSD-2-Clause-Patent.html
 
-   TERMS AND CONDITIONS FOR USE, REPRODUCTION, AND DISTRIBUTION
+Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
 
-   1. Definitions.
+1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
```
Comment (johannlombardi) [inline LICENSE:5]: Well, it explicitly mentions "the above copyright" that was removed. It sounds like we need to add it back then.

**Review by johannlombardi — CHANGES_REQUESTED** [review]
The license file isn't correct, i am afraid.
Please find an example of a project using the same license:
https://github.com/tianocore/edk2

File: LICENSE (hunk 2 above)
Comment (GitHuaKuang) [reply LICENSE]: Updated the LICENSE file, please take one more look see if it looks better.

**Comment by GitHuaKuang** [comment]
> The license file isn't correct, i am afraid.
> Please find an example of a project using the same license:
> https://github.com/tianocore/edk2

Updated accordingly.

File: LICENSE [hunk 3]
```diff
@@ -1,176 +1,51 @@
...
-      License. However, in accepting such obligations, You may act only
-      on Your own behalf and on Your sole responsibility, not on behalf
-      of any other Contributor, and only if You agree to indemnify,
-      defend, and hold each Contributor harmless for any liability
-      incurred by, or claims asserted against, such Contributor by reason
-      of your accepting any such warranty or additional liability.
-
-   END OF TERMS AND CONDITIONS
+Copyright (c) 2021, Intel Corporation.
+
```
Comment (jolivier23) [inline LICENSE:2]: Remove the copyright

**Review by jolivier23 — DISMISSED** [review]

File: LICENSE (hunk 3 above)
Comment (GitHuaKuang) [reply LICENSE:2]: As @johannlombardi pointed out that the content of this license actually mentioned the Copyright:

1. Redistributions of source code must retain the above copyright notice,

So we think we should provide copyright info.

**Review by johannlombardi — APPROVED** [review]

File: LICENSE (hunk 3 above)
Comment (jolivier23) [reply LICENSE:2]: the comma is not needed

Comment (jolivier23) [reply LICENSE:2]: Is the copyright just on this license file or on DAOS?   If it's on DAOS, it should be 2012-2021, I think.

**Review by jolivier23 — APPROVED** [review]

**Comment by GitHuaKuang** [comment]
@kjacque @johannlombardi @dpquigl Can I get approval one more time in order to land it? thx.

File: LICENSE (hunk 3 above)
Comment (GitHuaKuang) [reply LICENSE:2]: It is for this LICENSE file only, other daos files are keeping the existing Copyrights, just updated the date to 2021 together with whatever the existing beginning year.

**Review by bfaccini — APPROVED** [review]
@GitHuaKuang did you update all these files by hand ?? ;-)

**Comment by GitHuaKuang** [comment]
> @GitHuaKuang did you update all these files by hand ?? ;-)

Haha, good question, I must have been very enjoyed constantly pressing certain keyboard keys!

**Review by cibervicho — APPROVED** [review]

**Comment by GitHuaKuang** [comment]
@daos-stack/daos-gatekeeper Can we land this one? Thx.

**Pull request merged**